.docker/Dockerfile-CUDA-Base

```diff
@@ -1,142 +1,149 @@
-    wget https://dl.bintray.com/boostorg/release/1.70.0/source/boost_1_70_0.tar.gz && tar xf boost_1_70_0.tar.gz && \
-    cd arrayfire && git checkout v3.7.1  && git submodule update --init --recursive && \
-    mkdir build && cd build && \
-    CXXFLAGS=-DOS_LNX cmake .. -DCMAKE_BUILD_TYPE=Release -DAF_BUILD_CPU=OFF -DAF_BUILD_OPENCL=OFF -DAF_BUILD_EXAMPLES=OFF -DBOOST_INCLUDEDIR=/tmp/boost_1_70_0 && \
-    make -j$(nproc) && make install
+RUN cd /tmp && \
+    git clone --branch v3.7.3 --depth 1 --recursive --shallow-submodules https://github.com/arrayfire/arrayfire.git && \
+    mkdir -p arrayfire/build && cd arrayfire/build && \
+    CXXFLAGS=-DOS_LNX cmake .. -DCMAKE_BUILD_TYPE=Release \
+                               -DCMAKE_INSTALL_PREFIX=/opt/arrayfire \
+                               -DAF_BUILD_CUDA=ON \
+                               -DAF_BUILD_CPU=OFF \
+                               -DAF_BUILD_OPENCL=OFF \
+                               -DAF_BUILD_EXAMPLES=OFF \
+                               -DAF_WITH_IMAGEIO=OFF \
+                               -DBUILD_TESTING=OFF \
+                               -DAF_BUILD_DOCS=OFF && \
+    make install -j$(nproc)
+
+
+FROM cuda_base_builder as cuda_kenlm
 # ==================================================================
-# python (for bindings)
+# KenLM https://github.com/kpu/kenlm
 # ------------------------------------------------------------------
-RUN PIP_INSTALL="python3 -m pip --no-cache-dir install --upgrade" && \
-    DEBIAN_FRONTEND=noninteractive $APT_INSTALL \
-        software-properties-common \
-        && \
-    add-apt-repository ppa:deadsnakes/ppa && \
-    apt-get update && \
+RUN cd /tmp && git clone https://github.com/kpu/kenlm.git && \
+    cd kenlm && git checkout 0c4dd4e && \
+    mkdir build && cd build && \
+    cmake .. -DCMAKE_BUILD_TYPE=Release \
+             -DCMAKE_INSTALL_PREFIX=/opt/kenlm \
+             -DCMAKE_POSITION_INDEPENDENT_CODE=ON && \
+    make install -j$(nproc)
+
+#############################################################################
+#                             FINAL IMAGE                                   #
+#############################################################################
+
+FROM cuda_base_builder as cuda_final
+
+RUN apt-get update && \
     DEBIAN_FRONTEND=noninteractive $APT_INSTALL \
-        python3.6 \
-        python3.6-dev \
-        && \
-    wget -O ~/get-pip.py \
-        https://bootstrap.pypa.io/get-pip.py && \
-    python3.6 ~/get-pip.py && \
-    ln -s /usr/bin/python3.6 /usr/local/bin/python3 && \
-    ln -s /usr/bin/python3.6 /usr/local/bin/python && \
-    $PIP_INSTALL \
-        setuptools \
-        && \
-    $PIP_INSTALL \
-        sox \
-        tqdm \
-        numpy
+        vim \
+        emacs \
+        nano \
+        htop \
+        # nccl: for flashlight
+        libnccl2 libnccl-dev libglfw3-dev \
+        # for libsndfile for ubuntu 18.04, 20.04
+        libopus-dev \
```

Comment:
Do we need this? It's not listed [here](https://github.com/facebookresearch/flashlight/blob/master/README.md#dependencies)